src/Rings/FreeAssociativeAlgebraIdeal.jl

```diff
@@ -44,7 +44,7 @@ Return the two-sided ideal of $R$ generated by $g$.
 """
 function ideal(R::FreeAssociativeAlgebra, g::Vector{<:FreeAssociativeAlgebraElem})
   @req all(x -> parent(x) == R, g) "parent mismatch"
-  return FreeAssociativeAlgebraIdeal(IdealGens(R, g))
+  return FreeAssociativeAlgebraIdeal(IdealGens(R, collect(elem_type(R),g)))
 end
```

Comment:
I think it would be better to fix the `IdealGens` constructor so that the next user of `IdealGens` doesn't run into this again. 

I'll make a PR for that.